### PR TITLE
New Port: Ferium

### PR DESCRIPTION
**Pull request by JLindop**
#### Description

Ferium (https://github.com/gorilla-devs/ferium) is a CLI tool for updating Minecraft mods and mod packs.

`sudo port lint --nitpick` complains that there are no RMD160/size checksums for the ~70-odd Cargo packages that Ferium depends on, but I assume this isn't an issue in practice given it would be impractical to calculate checksums for them all.
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.5 23F79 arm64
Xcode 15.4 15F31d
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: games/ferium/Portfile [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+categories          games
+homepage            https://github.com/gorilla-devs/ferium
+license             MPL-2
+
+checksums           v${version}.tar.gz \
+                    rmd160 2cf1d569e05462e80839b9b9e1b67663ba9e30c3 \
+                    sha256 a7804e44d1949e3fe2c5e875d7d70fc6d2ecd16cbcb65619b9ad4f61ac3b1887 \
+                    size 6899654
+
+master_sites        https://github.com/gorilla-devs/ferium/archive/refs/tags/
```
Comment (herbygillot) [inline games/ferium/Portfile:25]: Thanks for this submission @JLindop.

This Portfile will greatly benefit from the `github` portgroup, which lets you use `github.setup` to not have to specify `master_sites`, `distfiles`, etc.  You can see an example of a `cargo` port using the `github` portgroup in various places, one example being `sysutils/hyperfine`.

File: games/ferium/Portfile [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+distfiles           v${version}.tar.gz
+
+use_configure       no
```
Comment (herbygillot) [inline games/ferium/Portfile:28]: You shouldn't have to specify this since you're using the `cargo` portgroup.

**Comment by herbygillot** [comment]
> sudo port lint --nitpick complains that there are no RMD160/size checksums for the ~70-odd Cargo packages that Ferium depends on, but I assume this isn't an issue in practice given it would be impractical to calculate checksums for them all.

This is correct.

File: games/ferium/Portfile (hunk 1 above)
Comment (JLindop) [reply games/ferium/Portfile:25]: I initially tried using the GitHub portgroup, but I couldn't get it to work. The source code isn't included in the GitHub releases, so I assume that I want `github.tarball_from tarball`, but the URL it generates (`https://codeload.github.com/gorilla-devs/ferium/legacy.tar.gz/v?dummy=`) always returns a 404.

This is how I configured the GitHub portgroup:
```tcl
PortGroup github    1.0

github.setup        gorilla-devs ferium v
github.tarball_from tarball
```

Comment (herbygillot) [reply games/ferium/Portfile:25]: You can try `archive` instead of `tarball`

Comment (herbygillot) [reply games/ferium/Portfile:25]: Here is documentation on the portgroup: https://guide.macports.org/chunked/reference.portgroup.html

Comment (JLindop) [reply games/ferium/Portfile:25]: I have seen that documentation, but perhaps I didn't understand it correctly.

Using `archive` generates the URL `https://github.com/gorilla-devs/ferium/archive/v`, but even if I manually correct it to `https://github.com/gorilla-devs/ferium/archive/v4.7.0`, it still returns a 404.

GitHub's [recommended URL for downloading tarballs by git tag](https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives) is `https://github.com/github/codeql/archive/refs/tags/<tag>`, but there seems to be no way to use that URL in the GitHub portgroup, which is why I set it manually.

**Comment by herbygillot** [comment]
Did you try `archive` while unsetting `distfile`?

`github.tarball_from` sets both `master_sites` and `distfiles`, while
relying on the version info from `github.setup`

**Comment by JLindop** [comment]
I got it working.

File: games/ferium/Portfile [hunk 3]
```diff
@@ -0,0 +1,334 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cargo 1.0
+PortGroup github    1.0
```
Comment (herbygillot) [inline games/ferium/Portfile:5]: Can you align this?

```suggestion
PortGroup           github    1.0
```

File: games/ferium/Portfile [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+maintainers         {johnlindop.com:git @JLindop} openmaintainer
+revision            0
+
+description         CLI program for downloading and updating Minecraft mods and modpacks.
+
+long_description    Ferium is a CLI tool for downloading and updating Minecraft mods from Modrinth \
+                    CurseForge, and GitHub Releases, and modpacks from Modrinth and CurseForge. \
+
+categories          games
+homepage            https://github.com/gorilla-devs/ferium
```
Comment (herbygillot) [inline games/ferium/Portfile:19]: `homepage` no longer needs to be explicitly set, since it is implicitly set by `github.setup`.

**Pull request merged**